### PR TITLE
Rust: Use incrementing versions in appstream

### PR DESCRIPTION
**Pull request by gasinvein**
Should fix both flathub/org.freedesktop.Sdk.Extension.rust-nightly#12 and flathub/org.freedesktop.Sdk.Extension.rust-nightly#14
Will set versions in appstream to likes `1.48.0` for stable and e.g. `1.50.0-nightly-20201224` for nightly, based on release date.
@tilosp @bilelmoussaoui Does this look valid to you?

### CODE REVIEW

File: src/checkers/rustchecker.py [hunk 1]
```diff
@@ -31,13 +36,21 @@ def check(self, external_data):
 
         package = data["pkg"][package_name]
         target = package["target"][target_name]
+
+        release_date = datetime.date.fromisoformat(data["date"])
+        version, _, _ = VERSION_RE.match(package["version"]).groups()
+        if channel == "stable":
```
Comment (tilosp) [inline src/checkers/rustchecker.py]: this case should probably include the `beta` channel
the versions look like this `1.49.0-beta.5 (b0dc3c6f4 2020-12-22)` so using `1.49.0-beta.5` should be fine

Comment (gasinvein) [reply src/checkers/rustchecker.py]: Exactly `beta` or any non-`nightly`?

Comment (tilosp) [reply src/checkers/rustchecker.py]: any non nightly seems best to me

**Review by tilosp — APPROVED** [review]
LGTM

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**